fog/ingest/server/src/controller.rs

```diff
@@ -265,7 +265,7 @@ where
     /// but with checks for idle state, and updates to sealed backups
     pub fn set_ingress_private_key(&self, msg: EnclaveMessage<PeerSession>) -> Result<(), Error> {
         // Lock our state for this entire call
-        let mut state = self.get_state();
+        let state = self.get_state();
```

Comment:
I'm not sure if `mut` was here by accident or if someone wanted to limit to only one reference at a time.
If it's the latter, I don't think it's necessary. My understanding is the MutexGuard is not Sync as such the compiler will make sure that only one thread would be accessing the `state`, after the `lock()` call.  So even if multiple locations, in this code was using the reference to the state, they shouldn't be fighting each other as they're on the same thread.

Reply:
It's probably just because `write_state_file_inner` wanted a `mut`, even if it didn't actually need one.